File: apps/codemap-search/tests/fixtures/navigation/cpp/policy.cpp
#include "policy.hpp"
#include <vector>

class OrderPolicy {
public:
    std::vector<std::string> evaluate(const Order &order) const {
        std::vector<std::string> failures;
        for (const auto &rule : rules_) {
            if (!rule.check(order)) {
                failures.push_back(rule.name());
            }
        }
        if (failures.empty()) {
            return {};
        }
        return failures;
    }

private:
    std::vector<Rule> rules_;
};

File: apps/codemap-search/tests/fixtures/navigation/rust/policy.rs
use std::collections::HashMap;

pub struct Policy<R> {
    rules: Vec<R>,
}

impl<R> Policy<R>
where
    R: Rule,
{
    pub fn evaluate(&self, order: &Order) -> HashMap<String, bool> {
        let failures: Vec<_> = self
            .rules
            .iter()
            .filter(|rule| !rule.check(order))
            .collect();

        if failures.is_empty() {
            return HashMap::new();
        }

        let mut outcomes = HashMap::new();
        for failure in &failures {
            outcomes.insert(failure.name().to_string(), false);
        }
        outcomes
    }
}
